src/react/components/pages/editorPage/assetPreview.tsx

```diff
@@ -59,6 +47,13 @@ export default class AssetPreview extends React.Component<IAssetPreviewProps, IA
         );
     }
 
+    private checkAssetPathProtocol(assetPath: string): string {
```

Comment:
Instead of having this check in the asset preview I'd recommend we push this logic back down into the underlying assetProvider to ensure the data is in a correct format before it is bound to the UI.  `IAssetProvider` has a `getAssets()` method.  I think this should be the location for this change.